### PR TITLE
Create doc about docker

### LINKED ISSUE #9: Create doc about Docker

### PR DESCRIPTION
**Pull request by rogatty**
Fixes #9

### CODE REVIEW

File: docker.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Classification / acceptance questions
+
+### Apprentice aka _I have a container running_
+* I can setup Docker on my system
+* I can run an image from a public repository
+
+### Intermediate aka _I have my app running in a container_
+* I know what a Docker image and container are
+* I can define a container with Dockerfile, use it to build the image and run it
```
Comment (gitkuba) [inline docker.md:22]: I think you define an image with Dockerfile, not container. 
From docker documentation:
> _Docker can build images automatically by reading the instructions from a Dockerfile_

File: docker.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+* I can map a container port to a different external port
+* I can publish an image to a repository
```
Comment (gitkuba) [inline docker.md:24]: I belive in docker terminology it's called registry.

File: docker.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+* I understand how an ingress routing mesh works
+* I know how to set up a private registry
+
+## Sources
+* [Docker docs](https://docs.docker.com/get-started/)
+* [The Difference Between Docker Compose And Docker Stack](https://vsupalov.com/difference-docker-compose-and-docker-stack/)
+* [Developer Tools Tutorials](https://github.com/docker/labs/blob/master/developer-tools/README.md)
+* [Docker Security](https://github.com/docker/labs/blob/master/security/README.md)
+
+## All custom stuff
```
Comment (gitkuba) [inline docker.md]: Redundant header

File: docker.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+* I can setup Docker on my system
+* I can run an image from a public repository
+
+### Intermediate aka _I have my app running in a container_
+* I know what a Docker image and container are
+* I can define a container with Dockerfile, use it to build the image and run it
+* I can map a container port to a different external port
+* I can publish an image to a repository
+
+### Advanced aka _I have my complex app running in multiple containers_
```
Comment (gitkuba) [inline docker.md:26]: Maybe mention good practicies for writing Dockerfiles?
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/

File: docker.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+* I know how to use multistage builds to keep images lean
+* I know how to set up a Swarm cluster and run an app in it
+
+### Master aka _?_
+* I know how to set up live debug for an app running inside a container
+* I know how to take advantage of Docker security features
+* I understand how an ingress routing mesh works
+* I know how to set up a private registry
+
+## Sources
```
Comment (gitkuba) [inline docker.md:40]: Videos from packt about docker are pretty good. I've learned a lot from: https://www.packtpub.com/application-development/getting-started-docker-video as i was approaching docker for the first time.

Comment (gitkuba) [reply docker.md]: Also available on our drive:  https://drive.google.com/drive/u/1/folders/1zo0kr69n6_uSsknO4ECZii9G7fL0YeOp

File: docker.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
+# Document title
+
+**Prerequisite knowledge**: web, server, infrastructure, virtualization, bash
```
Comment (kvas-damian) [inline docker.md]: In general, I don't think *web* is required to understand and learn Docker. For example, you can use docker for BI or machine learning purposes.

File: docker.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+# Document title
```
Comment (kvas-damian) [inline docker.md]: Please use a real title

File: docker.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+* I can publish an image to a repository
+
+### Advanced aka _I have my complex app running in multiple containers_
+* I can use docker-compose file to define multi-layered services
+* I know when to use Docker Stack and when Docker Compose
+* I know how to manage application data using volumes and bind mounts
+* I know how to use multistage builds to keep images lean
+* I know how to set up a Swarm cluster and run an app in it
+
+### Master aka _?_
```
Comment (kvas-damian) [inline docker.md]: Maybe "Docker hacker"?

Comment (rogatty) [reply docker.md]: `Ha(do)cker` 😛

File: docker.md (hunk 1 above)
Comment (rogatty) [reply docker.md:22]: You define container and use it to build the image :) See https://docs.docker.com/get-started/part2/#define-a-container-with-dockerfile

File: docker.md (hunk 2 above)
Comment (rogatty) [reply docker.md:24]: Registry is a collection of repositories and a repository is a collection of images, see https://docs.docker.com/get-started/part2/#share-your-image

File: docker.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+### Apprentice aka _I have a container running_
+* I can setup Docker on my system
+* I can run an image from a public repository
+
+### Intermediate aka _I have my app running in a container_
+* I know what a Docker image and container are
+* I can define a container with Dockerfile, use it to build the image and run it
+* I can map a container port to a different external port
+* I can publish an image to a repository
```
Comment (bognix) [inline docker.md:24]: You mentioned mapping ports, maybe you can add a point about mounting a volumen and what is a difference between volumen and using `ADD __file__` in Dockerfile.

Comment (rogatty) [reply docker.md:24]: @bognix in the Advanced section, there is:
>I know how to manage application data using volumes and bind mounts

Isn't this the same thing? Do you think I should move it to Intermediate?

File: docker.md (hunk 8 above)
Comment (gitkuba) [reply docker.md]: _I keep everything in containers... including my lunch_ 😆

File: docker.md [hunk 10]
```diff
@@ -28,19 +28,22 @@ Images are often created by combining and modifying standard images downloaded f
 * I know when to use Docker Stack and when Docker Compose
 * I know how to manage application data using volumes and bind mounts
 * I know how to use multistage builds to keep images lean
+* I know best practices for writing Dockerfiles
 * I know how to set up a Swarm cluster and run an app in it
 
-### Master aka _?_
+### Master aka _Ha(do)cker_
```
Comment (kvas-damian) [inline docker.md:34]: This one reminds me street fighter: https://streetfighter.fandom.com/wiki/Hadoken

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**